mm2src/coins/lightning/ln_utils.rs

```diff
@@ -129,47 +112,139 @@ fn netaddress_from_ipaddr(addr: IpAddr, port: u16) -> Vec<NetAddress> {
-    }
-    // Initialize the FeeEstimator. rpc_client implements the FeeEstimator trait, so it'll act as our fee estimator.
-    let fee_estimator = Arc::new(conf.rpc_client.clone());
+#[cfg(target_arch = "wasm32")]
+pub async fn start_lightning(
+    _ctx: &MmArc,
+    _platform_coin: UtxoStandardCoin,
+    _ticker: String,
+    _params: LightningActivationParams,
+    _network: Network,
+) -> EnableLightningResult<LightningCoin> {
+    MmError::err(EnableLightningError::UnsupportedMode(
+        "'connect_to_lightning_node'".into(),
+        "native".into(),
+    ))
+}
+
+#[cfg(not(target_arch = "wasm32"))]
+pub async fn start_lightning(
+    ctx: &MmArc,
+    platform_coin: UtxoStandardCoin,
+    ticker: String,
+    params: LightningActivationParams,
+    network: Network,
+) -> EnableLightningResult<LightningCoin> {
+    // The set (possibly empty) of socket addresses on which this node accepts incoming connections.
+    // If the user wishes to preserve privacy, addresses should likely contain only Tor Onion addresses.
+    let listening_addr = myipaddr(ctx.clone())
+        .await
+        .map_to_mm(EnableLightningError::InvalidAddress)?;
+    // If the listening port is used start_lightning should return an error early
+    let listener = TcpListener::bind(format!("{}:{}", listening_addr, params.listening_port))
+        .await
+        .map_to_mm(|e| EnableLightningError::IOError(e.to_string()))?;
+
+    // Initialize the FeeEstimator. UtxoStandardCoin implements the FeeEstimator trait, so it'll act as our fee estimator.
+    let fee_estimator = Arc::new(platform_coin.clone());
 
     // Initialize the Logger
     let logger = ctx.log.clone();
 
-    // Initialize the BroadcasterInterface. rpc_client implements the BroadcasterInterface trait, so it'll act as our transaction
+    // Initialize the BroadcasterInterface. UtxoStandardCoin implements the BroadcasterInterface trait, so it'll act as our transaction
     // broadcaster.
-    let broadcaster = Arc::new(conf.rpc_client.clone());
+    let broadcaster = Arc::new(platform_coin.clone());
 
     // Initialize Persist
-    let ln_data_dir = my_ln_data_dir(ctx)
+    let ln_data_dir = my_ln_data_dir(&ctx, &ticker)
         .as_path()
         .to_str()
         .ok_or("Data dir is a non-UTF-8 string")
         .map_to_mm(|e| EnableLightningError::InvalidPath(e.into()))?
         .to_string();
     let persister = Arc::new(FilesystemPersister::new(ln_data_dir.clone()));
 
-    // Initialize the Filter. rpc_client implements the Filter trait, so it'll act as our filter.
-    let filter = Some(Arc::new(conf.rpc_client.clone()));
+    // Initialize the Filter. PlatformFields implements the Filter trait, so it'll act as our filter.
+    let platform_fields = Arc::new(PlatformFields {
+        platform_coin,
+        registered_txs: AsyncMutex::new(HashMap::new()),
+        registered_outputs: AsyncMutex::new(Vec::new()),
+    });
+    let filter = Some(platform_fields.clone());
```

Comment:
This will change anyways when Native client is implemented for lightning. As replied here https://github.com/KomodoPlatform/atomicDEX-API/pull/1133#discussion_r759332593